sql/log.cc

```diff
@@ -5513,8 +5493,7 @@ int MYSQL_BIN_LOG::new_file_impl()
     last_used_log_number--;
   }
 
-end2:
-  if (delay_close)
+  if (!is_relay_log)
```

Comment:
And this also changes behaviour.

For example, in the previous code:
`is_relay_log == false`
`(error= write_event(&r)) != 0`

In this case, delay_close would remain false. With your patch, the file is always closed. That is not what the code did before. Is that acceptable or not? I do not know, but because of this, this is not a cleanup, rather a logic change.